Zend/zend_API.h

```diff
@@ -1954,21 +1954,24 @@ static zend_always_inline int zend_parse_arg_str_or_long(zval *arg, zend_string
 	return 1;
 }
 
-static zend_always_inline int zend_parse_arg_str_or_obj(
-	zval *arg, zend_string **dest_str, zend_object **dest_object, int allow_null
+static zend_always_inline int zend_parse_arg_class_name_or_obj(
+	zval *arg, zend_string **dest_str, zend_object **dest_object, int num, int allow_null
```

Comment:
We should make this `class_or_obj` instead with `zend_class_entry **` destination.

Reply:
In particular because that will save a duplicate zend_lookup_class, which is not exactly cheap.

Reply:
Yes, I realized this issue when I applied the macro in `get_parent_class()`, but before fixing it, I wanted to be sure that using an usual ZPP-mechanism is ok in this case :)

Reply:
At least for get_parent_class() the behavior looks pretty sensible to me. We also already have Z_PARAM_CLASS, so it should fit in well with existing semantics.